climada/hazard/base.py

```diff
@@ -773,6 +798,10 @@ def select(self, event_names=None, event_id=None, date=None, orig=None,
 
         # reset frequency if date span has changed (optional):
         if reset_frequency:
+            if self.frequency_unit not in ['1/year', 'annual', '1/y', '1/a']:
+                LOGGER.warning("resetting the frequency on a hazard object who's frequency unit"
+                    "is %s and not %s will most likely lead to unexpected results",
```

Comment:
I do not really like the "will lead to unexpected results". This very vague. Can we make it more precise you think?

Reply:
Good question! Can we? The thing is that the user is sending contradicting signals at this point. Why do they set a frequency_unit other than 'annual' and then still use the method in a way that is strongly related to 'annual' frequencies.
I'd say they have to look and see themselves. 🤷

Reply:
I think we could say something like:

"The frequency unit will be changed from %s to %s. The resulting frequency will thus be most likely incorrect. Consider setting reset_frequency=False and set the frequency manually for the selection."